ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala

```diff
@@ -87,16 +92,18 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
     water.DKV.put(trainFrame)
     
     // Train
-    val model: M with H2OModelParams = trainModel(getParams)
+
+    val binaryModel = trainModel(getParams)
+    val model = new H2OMOJOModel(ModelSerializationSupport.getMojoData(binaryModel))
```

Comment:
Here should be `val model: M with H2OModelParams`, unless the methods `setFeaturesCols` etc should not be visible, or am I missing something?

Reply:
This should be solved by removing ```M``` from the generic parameters.